### PR TITLE
[FLINK-31292] Introduce HadoopUtils to get Configuration in CatalogContext

### PR DESCRIPTION
**Pull request by liyubin117**
At present, if HadoopConf is not passed in the CatalogContext, a new HadoopConf will be directly generated, which may not have the required parameters. we should correct it.

### CODE REVIEW

File: flink-table-store-common/src/main/java/org/apache/flink/table/store/utils/HadoopConfUtils.java [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+ * Utility class for working with Hadoop-related classes. This should only be used if Hadoop is on
+ * the classpath.
+ */
+public class HadoopConfUtils {
+
+    private static final Logger LOG = LoggerFactory.getLogger(HadoopConfUtils.class);
+    public static final String HADOOP_HOME_ENV = "HADOOP_HOME";
+    public static final String HADOOP_CONF_ENV = "HADOOP_CONF_DIR";
+
+    public static Configuration getHadoopConfiguration() {
```
Comment (JingsongLi) [inline flink-table-store-common/src/main/java/org/apache/flink/table/store/utils/HadoopConfUtils.java:38]: We can pass `Options` in here to support some config options like Flink HadoopUtils.

Comment (liyubin117) [reply flink-table-store-common/src/main/java/org/apache/flink/table/store/utils/HadoopConfUtils.java:38]: Thanks for your reminds, I wonder if it is reasonable to get extra options like Flink HadoopUtils, as of the feature in common module invoked by Spark/Hive should not rely on Fllink configuration.

Comment (JingsongLi) [reply flink-table-store-common/src/main/java/org/apache/flink/table/store/utils/HadoopConfUtils.java:38]: I think these should be our configuration, `Approach 2: Flink configuration (deprecated)` is deprecated, but approach 4 may looks good:
```
// Approach 4: catalog options
// add all configuration key with prefix 'hadoop.' in catalog options to hadoop conf
```
Maybe like this?

Comment (liyubin117) [reply flink-table-store-common/src/main/java/org/apache/flink/table/store/utils/HadoopConfUtils.java:38]: I get it~ to make it clear, do you mean make it like `create(options, HadoopConfUtils.getHadoopConfiguration(options))`

Comment (JingsongLi) [reply flink-table-store-common/src/main/java/org/apache/flink/table/store/utils/HadoopConfUtils.java:38]: Yes~

File: flink-table-store-common/src/main/java/org/apache/flink/table/store/utils/HadoopUtils.java [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+import java.io.File;
+
+/**
+ * Utility class for working with Hadoop-related classes. This should only be used if Hadoop is on
+ * the classpath. Note: decoupled from specific engines.
+ */
+public class HadoopUtils {
+
+    private static final Logger LOG = LoggerFactory.getLogger(HadoopUtils.class);
+    private static final String[] FLINK_CONFIG_PREFIXES = {"flink.hadoop."};
```
Comment (JingsongLi) [inline flink-table-store-common/src/main/java/org/apache/flink/table/store/utils/HadoopUtils.java:37]: Don't use `flink.`.
`String[] FLINK_CONFIG_PREFIXES = {"flink.hadoop."}` -> `String[] CONFIG_PREFIXES = {"hadoop."}`.

Comment (liyubin117) [reply flink-table-store-common/src/main/java/org/apache/flink/table/store/utils/HadoopUtils.java]: done :)

**Review by JingsongLi — COMMENTED** [review]
Can you add more tests, you can refer to Flink `HadoopConfigLoadingTest`.

**Comment by liyubin117** [comment]
@JingsongLi I have added more tests, please take a look

**Review by JingsongLi — APPROVED** [review]
+1

**Pull request merged**